README.rst

```diff
@@ -385,51 +417,209 @@ Add ``openwisp_utils.loaders.DependencyLoader`` to ``TEMPLATES`` in your ``setti
+
+For example:
+
+.. code-block:: python
+
+    from openwisp_firmware_upgrader.admin import (  # noqa
+        BatchUpgradeOperationAdmin,
+        BuildAdmin,
+        CategoryAdmin,
+    )
+
+    BuildAdmin.list_display.insert(1, 'my_custom_field')
+    BuildAdmin.ordering = ['-my_custom_field']
+
+2. Inheriting admin classes
+###########################
+
+If you need to introduce significant changes and/or you don't want to resort to
+monkey patching, you can proceed as follows:
+
+.. code-block:: python
+
+    from django.contrib import admin
+    from openwisp_firmware_upgrader.admin import (
+        BatchUpgradeOperationAdmin as BaseBatchUpgradeOperationAdmin,
+        BuildAdmin as BaseBuildAdmin,
+        CategoryAdmin as BaseCategoryAdmin,
+    )
+    from openwisp_firmware_upgrader.swapper import load_model
+
+    BatchUpgradeOperation = load_model('BatchUpgradeOperation')
+    Build = load_model('Build')
+    Category = load_model('Category')
+    DeviceFirmware = load_model('DeviceFirmware')
+    FirmwareImage = load_model('FirmwareImage')
+    UpgradeOperation = load_model('UpgradeOperation')
+
+    admin.site.unregister(BatchUpgradeOperation)
+    admin.site.unregister(Build)
+    admin.site.unregister(Category)
+
+    class BatchUpgradeOperationAdmin(BaseBatchUpgradeOperationAdmin):
+        # add your changes here
+
+    class BuildAdmin(BaseBuildAdmin):
+        # add your changes here
+
+    class CategoryAdmin(BaseCategoryAdmin):
+        # add your changes here
+
+11. Create root URL configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Please refer to the `urls.py <https://github.com/openwisp/openwisp-firmware-upgrader/tree/master/tests/openwisp2/urls.py>`_
+file in the test project.
+
+For more information about URL configuration in django, please refer to the
+`"URL dispatcher" section in the django documentation <https://docs.djangoproject.com/en/dev/topics/http/urls/>`_.
+
+12. Create celery.py
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Please refer to the `celery.py <https://github.com/openwisp/openwisp-firmware-upgrader/tree/master/tests/openwisp2/celery.py>`_
+file in the test project.
+
+For more information about the usage of celery in django, please refer to the
+`"First steps with Django" section in the celery documentation <https://docs.celeryproject.org/en/master/django/first-steps-with-django.html>`_.
+
+13. Import the automated tests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 When developing a custom application based on this module, it's a good
-idea to import and run the `base tests <https://github.com/openwisp/openwisp-firmware-upgrader/tree/master/openwisp_firmware_upgrader/tests/base>`_
-too, so that you can be sure the changes you're introducing are not breaking
-some of the existing features of openwisp-firmware-upgrader.
+idea to import and run the base tests too, so that you can be sure the changes
+you're introducing are not breaking some of the existing features of openwisp-firmware-upgrader.
```

Comment:
Just to point out a small detail, everywhere else, you've used `*openwisp-firmware-upgrader*`, here it's only `openwisp-firmware-upgrader`. :smile: